RFS/src/main/java/com/rfs/common/DocumentReindexer.java

```diff
@@ -11,34 +12,53 @@
 import lombok.RequiredArgsConstructor;
 import reactor.core.publisher.Flux;
 import reactor.core.publisher.Mono;
+import reactor.core.scheduler.Schedulers;
 
 @RequiredArgsConstructor
 public class DocumentReindexer {
     private static final Logger logger = LogManager.getLogger(DocumentReindexer.class);
     protected final OpenSearchClient client;
     private final int numDocsPerBulkRequest;
-    private final int maxConcurrentRequests;
+    private final long numBytesPerBulkRequest;
+    private final int maxConcurrentWorkItems;
 
     public Mono<Void> reindex(
         String indexName,
         Flux<Document> documentStream,
         IDocumentMigrationContexts.IDocumentReindexContext context
     ) {
-
-        return documentStream.map(this::convertDocumentToBulkSection)  // Convert each Document to part of a bulk
-                                                                       // operation
-            .buffer(numDocsPerBulkRequest) // Collect until you hit the batch size
-            .doOnNext(bulk -> logger.info("{} documents in current bulk request", bulk.size()))
-            .map(this::convertToBulkRequestBody)  // Assemble the bulk request body from the parts
+        var documentStreamRunner = Schedulers.newSingle("luceneRunner");
+        var requestSendingRunner = Schedulers.newParallel("requestSendingRunner", maxConcurrentWorkItems);
+        return documentStream
+            .subscribeOn(documentStreamRunner)
+            .map(this::convertDocumentToBulkSection)
+            .subscribeOn(documentStreamRunner) // Use same thread for lucene reading and buffering
```

Comment:
what happens if you leave this out?

Reply:
I didn't need both `subscribeOn(documentStreamRunner)`. I removed that.

I also did some testing by removing all specified `.subscribeOn`, `.publishOn`, `runOn` calls and with 2cpu4gb fargate throughput went from 245 -> 474 10MB requests/min.

I'll keep these out and once we have the infrastructure to do performance testing, we can change it up